### PR TITLE
[closes #36] Fix MavenCli for HPUX compatibility

### LINKED ISSUE #36: Fix starting maven process on HPUX

### PR DESCRIPTION
**Pull request by mswiech**

### CODE REVIEW

File: src/main/java/org/jboss/qa/jenkins/test/executor/utils/SyncProcessRunner.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		err.start();
+
+		final int result = process.waitFor();
+		out.join();
+		err.join();
+		return result;
+	}
+
+	/**
+	 * This is thread is workaround, because processBuilder.redirectOutput(ProcessBuilder.Redirect.INHERIT)
```
Comment (basovnik) [inline src/main/java/org/jboss/qa/jenkins/test/executor/utils/SyncProcessRunner.java]: Typo "is".

File: src/main/java/org/jboss/qa/jenkins/test/executor/utils/MavenCli.java [hunk 2]
```diff
@@ -417,10 +415,12 @@ public Builder params(Collection<String> params) {
 
 		public MavenCli build() {
 			if (mavenHome != null && !mavenHome.exists()) {
-				throw new IllegalArgumentException(String.format("Maven home does not exist: %s", mavenHome.getAbsolutePath()));
+				throw new IllegalArgumentException(String.format("Maven home does not exist: %s",
```
Comment (VaclavChalupa) [inline src/main/java/org/jboss/qa/jenkins/test/executor/utils/MavenCli.java]: @mswiech please do not split unchanged lines - there is no checkstyle restriction :)

File: src/main/java/org/jboss/qa/jenkins/test/executor/utils/MavenCli.java [hunk 3]
```diff
@@ -417,10 +415,12 @@ public Builder params(Collection<String> params) {
...
+						mavenHome.getAbsolutePath()));
 			}
 			if (javaHome != null && !javaHome.exists()) {
-				throw new IllegalArgumentException(String.format("Java home does not exist: %s", javaHome.getAbsolutePath()));
+				throw new IllegalArgumentException(String.format("Java home does not exist: %s",
```
Comment (VaclavChalupa) [inline src/main/java/org/jboss/qa/jenkins/test/executor/utils/MavenCli.java]: @mswiech please do not split unchanged lines - there is no checkstyle restriction :)

File: src/main/java/org/jboss/qa/jenkins/test/executor/utils/SyncProcessRunner.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+			this.forErr = forErr;
+			this.process = process;
+			this.print = forErr ? System.err : System.out;
+			this.setDaemon(true);
+		}
+
+		@Override
+		public void run() {
+			try (BufferedReader r = new BufferedReader(
+					new InputStreamReader(forErr ? process.getErrorStream() : process.getInputStream()))) {
```
Comment (VaclavChalupa) [inline src/main/java/org/jboss/qa/jenkins/test/executor/utils/SyncProcessRunner.java]: @mswiech there is no LineLength checkstyle rule, you can keep it on one line ;)

File: src/main/java/org/jboss/qa/jenkins/test/executor/utils/MavenCli.java [hunk 5]
```diff
@@ -238,10 +238,9 @@ public boolean isFailAtEnd() {
 	}
 
 	public int run() throws Exception {
-
 		final ProcessBuilder processBuilder = new ProcessBuilder(generateCommand());
-		processBuilder.redirectOutput(ProcessBuilder.Redirect.INHERIT);
-		processBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
+		//processBuilder.redirectOutput(ProcessBuilder.Redirect.INHERIT) doesn't work on HPUX.
```
Comment (VaclavChalupa) [inline src/main/java/org/jboss/qa/jenkins/test/executor/utils/MavenCli.java]: @mswiech please remove from here, put all info to javadoc of RedirectProcessOut.

File: src/main/java/org/jboss/qa/jenkins/test/executor/utils/SyncProcessRunner.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	 * This thread is workaround, because processBuilder.redirectOutput(ProcessBuilder.Redirect.INHERIT)
+	 * doesn't work on HPUX.
+	 */
+	public class RedirectProcessOut extends Thread {
+
+		private final boolean forErr;
+		private final Process process;
+		private final PrintStream print;
+
+		public RedirectProcessOut(final Process process, final boolean forErr) {
```
Comment (VaclavChalupa) [inline src/main/java/org/jboss/qa/jenkins/test/executor/utils/SyncProcessRunner.java]: @mswiech please choose a better name for class, ProcessOutputConsumer or something like that.

File: src/main/java/org/jboss/qa/jenkins/test/executor/utils/SyncProcessRunner.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+			this.forErr = forErr;
+			this.process = process;
+			this.print = forErr ? System.err : System.out;
```
Comment (VaclavChalupa) [inline src/main/java/org/jboss/qa/jenkins/test/executor/utils/SyncProcessRunner.java]: @mswiech What about to send directly the PrintStream? - change signature of constructor.

**Comment by mswiech** [comment]
all comments processed

**Pull request merged**